schedule.js

```diff
@@ -13,12 +13,24 @@ import CGUs from './src/index.js';
 
     const app = new CGUs();
     await app.init();
-
     const notifier = new Notifier(app.serviceDeclarations, app.documentTypes);
 
-    app.on('versionRecorded', notifier.onVersionRecorded.bind(notifier));
-    app.on('fetchingError', notifier.onDocumentFetchError.bind(notifier));
-    app.on('error', notifier.onApplicationError.bind(notifier));
+    const delayedVersionNotitificationsParams = [];
+
+    app.on('versionRecorded', (serviceId, type, versionId) => {
+      delayedVersionNotitificationsParams.push({ serviceId, type, versionId });
+    });
+
+    app.on('changesPublished', () => {
+      delayedVersionNotitificationsParams.forEach(({ serviceId, type, versionId }) => {
+        notifier.onVersionRecorded(serviceId, type, versionId);
```

Comment:
> Rename the function, since it is not really a handler anymore?

Reply:
ok, in this case, I also rename the other exposed functions for consistency: 
`onVersionRecorded` -> `notifyVersionRecorded` or `versionRecorded`
`onDocumentFetchError` -> `documentFetchError` or …
`onDocumentUpdateError` -> `documentUpdateError` or …
`onApplicationError` -> `applicationError` or …
?

Reply:
Never mind, I didn't properly take into account the file… I somehow thought we were in the notifier itself, not in the caller. Sorry, this comment is not appropriate 😞